server/src/controllers/Events.js

```diff
@@ -79,8 +86,15 @@ module.exports = {
     const eventId = req.params.id;
     const event = await Event.findById(eventId);
     await event.remove();
-    await cloudinary.v2.uploader.destroy(eventId);
-    res.status(204);
+    try {
+      await cloudinary.api.resource(eventId);
+      try {
+        await cloudinary.v2.uploader.destroy(eventId);
+      } catch(error) {
+        res.status(500).json({});
+      }
+    } catch(error) {}
```

Comment:
This change is added as the first try-catch block checks whether the image exists or not and the nested try-catch block checks if a server-side error occurred during the deletion of image.